src/Coordination/KeeperServer.cpp

```diff
@@ -990,7 +991,7 @@ KeeperServer::ConfigUpdateState KeeperServer::applyConfigUpdate(
         raft_instance->set_priority(update->id, update->priority, /*broadcast on live leader*/true);
         return Accepted;
     }
-    UNREACHABLE();
+    throw Exception(ErrorCodes::UNEXPECTED_ZOOKEEPER_ERROR, "Unexpected action");
```

Comment:
Same here, as in `KeeperReconfiguration.cpp`: I would use `chassert()` to make sure keeper crashes.

Reply:
Everything that is called from inside the Raft is safer to abort instead of throwing exception.
This should be okay but `KeeperDispatcher::clusterUpdateThread` is missing a try/catch so it could lead to uncaught exceptions.